packages/mui-system/src/createStyled.js

```diff
@@ -135,7 +135,15 @@ export default function createStyled(input = {}) {
           const styleOverrides = getStyleOverrides(componentName, theme);
 
           if (styleOverrides) {
-            return overridesResolver(props, styleOverrides);
+            const parsedStyleOverrides = {};
+            Object.entries(styleOverrides).forEach(([slot, styles]) => {
+              if (typeof styles === 'function') {
+                parsedStyleOverrides[slot] = styles({ ...props, ...props.ownerState, theme });
```

Comment:
`ownerState` is not redundant. It is for the slots. Take `Chip` as an example:

```js
<ChipRoot
  as={component}
  className={clsx(classes.root, className)}
  disabled={clickable && disabled ? true : undefined}
  onClick={onClick}
  onKeyDown={handleKeyDown}
  onKeyUp={handleKeyUp}
  ref={handleRef}
  ownerState={ownerState}
  {...moreProps}
  {...other}
>
  {avatar || icon}
  <ChipLabel className={clsx(classes.label)} ownerState={ownerState}>
    {label}
  </ChipLabel>
  {deleteIcon}
</ChipRoot>
```

We need to spread `props.ownerState`, so that developer can use it in the callback for styling `ChipLabel`.

Reply:
In that case, do not spread it. It is confusing when you should use it, and when it is not required. It's better to always depend on it.

Reply:
Especially for the slots there may be conflicts in terms of props and `ownerState` values, I propose we do not introduce this change until we are sure it works and validate that it works in every case.

Reply:
> do not spread it

If we don't spread `ownerState`, how to style slots based on props? Also dealing with Typescript will be tough work because you have to specify which key is Root and which keys are slots.

> conflicts in terms of props

Is this enough > `{ ...props.ownerState, ...props }` so the slot's props does not accidentally overridden by the ownerState.

> until we are sure it works and validate that it works in every case

What I can think of is add a test in `describeConformance` to make sure that all of the slots in every components receive ownerState.

Reply:
> If we don't spread ownerState, how to style slots based on props? Also dealing with Typescript will be tough work because you have to specify which key is Root and which keys are slots.

Developers can use `ownerState` as we do in the slot components, no? I don't understand the problem.

Reply:
> Developers can use ownerState as we do in the slot components

Yes, we can do that (I thought we meant `ownerState` to be private but I can see that it is safer than spreading it).

I will also update the types so that slots has `ownerState` field in the props and add it to the docs.

Reply:
@mnajdova Turns out there is no change on the implementation, just adding callback types.